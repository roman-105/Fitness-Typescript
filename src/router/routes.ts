export enum Routes {
  Home = 'HOME',
  Programs = 'EXPLORE',
  Progress = 'PROGRESS',
  Coach = 'COACH',
  Club = 'CLUBS',
  Main = 'MAIN',
  Profile = 'PROFILE',
  QR = 'QR',
  Notifications = 'NOTIFICATIONS',
  Search = 'SEARCH',
  Settings = 'SETTINGS',
  Webview = 'WEBVIEW',
  Login = 'LOGIN',
  SignUp = 'SIGNUP',
  CastController = 'CAST-CONTROLLER',
  ClubSingular = 'CLUB-SINGULAR',
  ClubFinder = 'CLUB-FINDER',
  LessonSchedule = 'LESSON-SCHEDULE',
  ClubWorkouts = 'CLUB-WORKOUTS',
  WorkoutSingular = 'WORKOUT-SINGULAR',
  AllClubWorkouts = 'ALL-CLUB-WORKOUTS',
  RecommendedClubWorkouts = 'RECOMMENDED-WORKOUTS',
  OnDemandClubWorkouts = 'ON-DEMAND-WORKOUTS',
  LessonSingular = 'LESSON-SINGULAR',
  Consent = 'CONSENT',
  Onboarding = 'ONBOARDING',
  OnboardingConfirm = 'ONBOARDING-CONFIRM',
  tests = 'TESTS'
}
