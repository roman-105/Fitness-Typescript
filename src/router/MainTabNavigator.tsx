import React, { useEffect, useCallback } from 'react';
import { createBottomTabNavigator } from '@react-navigation/bottom-tabs';
import { useNavigation } from '@react-navigation/native';
import { bottomBarIcon, bottomBarLabel } from '../components/BottomBarIcon';
import { HeaderContainer } from '../components/Header';
import theme from '../theme';
import { Routes } from './routes';

import HomeScreen from '../screens/home/Home';
import ClubScreen from '../screens/club/Club';
import ExploreScreen from '../screens/explore/Explore';
import ProgressScreen from '../screens/progress/Progress';
import CoachScreen from '../screens/coach/Coach';
import usePrivate from '../hooks/usePrivate';

const Tab = createBottomTabNavigator();

const headerConfigLookup: any = {
  [Routes.Home]: (navigateTo: (route: Routes) => void) => ({
    route: Routes.Home,
    left: [{ icon: 'profile', onPress: () => navigateTo(Routes.Profile) }],
    right: [
      { icon: 'card', onPress: () => navigateTo(Routes.QR) },
      { icon: 'notifications', onPress: () => navigateTo(Routes.Notifications) }
    ]
  }),
  [Routes.Programs]: (navigateTo: (route: Routes) => void) => ({
    route: Routes.Programs,
    left: [{ icon: 'profile', onPress: () => navigateTo(Routes.Profile) }],
    right: [
      { icon: 'search', onPress: () => navigateTo(Routes.Search) },
      { icon: 'notifications', onPress: () => navigateTo(Routes.Notifications) }
    ]
  }),
  [Routes.Progress]: (navigateTo: (route: Routes) => void) => ({
    route: Routes.Progress,
    left: [{ icon: 'profile', onPress: () => navigateTo(Routes.Profile) }],
    right: [
      { icon: 'search', onPress: () => navigateTo(Routes.Search) },
      { icon: 'notifications', onPress: () => navigateTo(Routes.Notifications) }
    ]
  }),
  [Routes.Coach]: (navigateTo: (route: Routes) => void) => ({
    route: Routes.Coach,
    left: [{ icon: 'profile', onPress: () => navigateTo(Routes.Profile) }],
    right: [
      { icon: 'search', onPress: () => navigateTo(Routes.Search) },
      { icon: 'notifications', onPress: () => navigateTo(Routes.Notifications) }
    ]
  }),
  [Routes.Club]: (navigateTo: (route: Routes) => void) => ({
    route: Routes.Club,
    left: [{ icon: 'profile', onPress: () => navigateTo(Routes.Profile) }],
    right: [
      { icon: 'card', onPress: () => navigateTo(Routes.QR) },
      { icon: 'notifications', onPress: () => navigateTo(Routes.Notifications) }
    ]
  })
};

const MainTabNavigator = ({
  Home = HomeScreen,
  Programs = ExploreScreen,
  Progress = ProgressScreen,
  Coach = CoachScreen,
  Club = ClubScreen
}) => {
  const navigation = useNavigation();
  const indexTab: number | undefined = navigation.dangerouslyGetState().routes[0]?.state?.index;

  const navigateTo = useCallback(
    (route: Routes) => {
      navigation.navigate(route);
    },
    [navigation]
  );

  useEffect(() => {
    const navState: any = navigation.dangerouslyGetState().routes[0]?.state;
    // indexTab === undefined => firstTab

    const headerConfig =
      indexTab !== undefined
        ? headerConfigLookup[navState?.routeNames[indexTab]](navigateTo)
        : headerConfigLookup[Routes.Home](navigateTo);

    navigation.setOptions({
      headerLeft: () => <HeaderContainer route={headerConfig.route} items={headerConfig.left} />,
      headerRight: () => <HeaderContainer route={headerConfig.route} items={headerConfig.right} />
    });
  }, [navigation, indexTab, navigateTo]);

  // make the whole tab navigator private
  usePrivate();

  return (
    <Tab.Navigator
      screenOptions={({ route }) => ({
        tabBarIcon: bottomBarIcon(route.name as Routes),
        tabBarLabel: bottomBarLabel(route.name as Routes)
      })}
      tabBarOptions={{
        style: { height: 48 },
        showLabel: true,
        inactiveTintColor: '#838386', // this color only appears here
        activeTintColor: theme.colors.primary.orange
      }}
    >
      <Tab.Screen options={{ tabBarTestID: 'home-tab' }} name={Routes.Home} component={Home} />
      <Tab.Screen
        options={{ tabBarTestID: 'programs-tab' }}
        name={Routes.Programs}
        component={Programs}
      />
      <Tab.Screen
        options={{ tabBarTestID: 'progress-tab' }}
        name={Routes.Progress}
        component={Progress}
      />
      <Tab.Screen options={{ tabBarTestID: 'coach-tab' }} name={Routes.Coach} component={Coach} />
      <Tab.Screen options={{ tabBarTestID: 'club-tab' }} name={Routes.Club} component={Club} />
    </Tab.Navigator>
  );
};

export default MainTabNavigator;
