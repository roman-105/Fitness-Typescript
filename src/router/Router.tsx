import React, { useEffect } from 'react';
import { StyleSheet, TouchableOpacity, StatusBar } from 'react-native';
import { createStackNavigator } from '@react-navigation/stack';
import { Routes } from './routes';
import MainTabNavigator from './MainTabNavigator';
import ProfileScreen from '../screens/profile/Profile';
import { HeaderContainer } from '../components/Header';
import theme from '../theme';
import NotificationsScreen from '../screens/notifications/Notifications';
import QRScreen from '../screens/club/QR';
import SettingsScreen from '../screens/profile/settings/Settings';
import SearchScreen from '../screens/search/Search';
import { ArrowLeftIcon } from '../components/Icon';
import BFWebview from '../components/Webview/BFWebview';
import TestsScreen from '../screens/tests/Tests';
import LoginScreen from '../screens/login/Login';
import SignUpScreen from '../screens/login/SignUp';
import { SafeAreaView } from 'react-native-safe-area-context';
import { CastController } from '../components/CastController';
import { useSelector, useDispatch } from 'react-redux';
import ClubSingular from '../screens/club/singular-club/ClubSingular';
import ClubFinder from '../screens/club/club-finder/ClubFinder';
import LessonsSingular from '../screens/lesson/singular-lesson/LessonSingular';
import Lesson from '../screens/lesson/Lesson';
import ClubWorkouts from '../screens/explore/club-workouts/ClubWorkouts';
import AllClubWorkouts from '../screens/explore/club-workouts/all-club-workouts/AllClubWorkouts';
import WorkoutSingular from '../screens/explore/singular-workout/WorkoutSingular';
import RecommendedClubWorkouts from '../screens/explore/club-workouts/recommended/RecommendedWorkouts';
import OnDemandClubWorkouts from '../screens/explore/club-workouts/on-demand/OnDemandWorkouts';
import ConsentScreen from '../screens/personalisation/consent/ConsentScreen';
import Personalisation from '../screens/personalisation/Personalisation';
import PersonalisationConfirmation from '../screens/personalisation/confirmation/PersonalisationConfirmation';
import formatMessage from 'format-message';

const styles: any = StyleSheet.create({
  headerContainer: {
    height: 48,
    backgroundColor: theme.colors.primary.white,
    shadowOpacity: 0,
    elevation: 0
  },
  backButton: {
    width: 40,
    justifyContent: 'center',
    alignItems: 'center'
  },
  safeAreaContainer: { flex: 1 }
});

const Stack = createStackNavigator();

function RouterComponent() {
  return (
    <Stack.Navigator
      screenOptions={{
        headerStyle: styles.headerContainer,
        headerLeftContainerStyle: { marginHorizontal: 12 },
        headerRightContainerStyle: { marginHorizontal: 12 },
        headerTitle: '',
        headerLeft: (config) => {
          return (
            <TouchableOpacity
              testID="header-back-button"
              style={styles.backButton}
              onPress={config.onPress}
            >
              <ArrowLeftIcon fill={theme.colors.black} />
            </TouchableOpacity>
          );
        }
      }}
    >
      <Stack.Screen name={Routes.Main} component={MainTabNavigator} />
      <Stack.Screen options={{ header: () => null }} name={Routes.Login} component={LoginScreen} />
      <Stack.Screen name={Routes.SignUp} component={SignUpScreen} />
      <Stack.Screen
        options={({ navigation }) => ({
          headerRight: () => {
            return (
              <HeaderContainer
                route={Routes.Profile}
                items={[
                  {
                    icon: 'settings',
                    onPress: () => navigation.navigate(Routes.Settings)
                  },
                  {
                    icon: 'notifications',
                    onPress: () => navigation.navigate(Routes.Notifications)
                  }
                ]}
              />
            );
          }
        })}
        name={Routes.Profile}
        component={ProfileScreen}
      />
      <Stack.Screen name={Routes.Notifications} component={NotificationsScreen} />
      <Stack.Screen name={Routes.QR} component={QRScreen} />
      <Stack.Screen name={Routes.Settings} component={SettingsScreen} />
      <Stack.Screen name={Routes.Search} component={SearchScreen} />
      <Stack.Screen name={Routes.Webview} component={BFWebview} />
      <Stack.Screen name={Routes.tests} component={TestsScreen} />
      <Stack.Screen name={Routes.ClubFinder} component={ClubFinder} />
      <Stack.Screen name={Routes.LessonSchedule} component={Lesson} />
      <Stack.Screen name={Routes.Consent} component={ConsentScreen} />
      <Stack.Screen name={Routes.Onboarding} component={Personalisation} />
      <Stack.Screen name={Routes.OnboardingConfirm} component={PersonalisationConfirmation} />

      <Stack.Screen
        options={{ headerShown: false }}
        name={Routes.LessonSingular}
        component={LessonsSingular}
      />
      <Stack.Screen
        options={{ headerShown: false }}
        name={Routes.ClubSingular}
        component={ClubSingular}
      />
      <Stack.Screen
        options={{ headerShown: false }}
        name={Routes.CastController}
        component={CastController}
      />
      <Stack.Screen
        options={({ navigation }) => ({
          headerRight: () => {
            return (
              <HeaderContainer
                route={Routes.ClubWorkouts}
                items={[
                  {
                    icon: 'search',
                    onPress: () => navigation.navigate(Routes.Search)
                  },
                  {
                    icon: 'notifications',
                    onPress: () => navigation.navigate(Routes.Notifications)
                  }
                ]}
              />
            );
          }
        })}
        name={Routes.ClubWorkouts}
        component={ClubWorkouts}
      />
      <Stack.Screen
        options={({ navigation }) => ({
          headerRight: () => {
            return (
              <HeaderContainer
                route={Routes.AllClubWorkouts}
                items={[
                  {
                    icon: 'search',
                    onPress: () => navigation.navigate(Routes.Search)
                  },
                  {
                    icon: 'notifications',
                    onPress: () => navigation.navigate(Routes.Notifications)
                  }
                ]}
              />
            );
          }
        })}
        name={Routes.AllClubWorkouts}
        component={AllClubWorkouts}
      />
      <Stack.Screen
        options={({ navigation }) => ({
          headerRight: () => {
            return (
              <HeaderContainer
                route={Routes.WorkoutSingular}
                items={[
                  {
                    icon: 'search',
                    onPress: () => navigation.navigate(Routes.Search)
                  },
                  {
                    icon: 'notifications',
                    onPress: () => navigation.navigate(Routes.Notifications)
                  }
                ]}
              />
            );
          }
        })}
        name={Routes.WorkoutSingular}
        component={WorkoutSingular}
      />
      <Stack.Screen
        options={({ navigation }) => ({
          headerRight: () => {
            return (
              <HeaderContainer
                route={Routes.RecommendedClubWorkouts}
                items={[
                  {
                    icon: 'search',
                    onPress: () => navigation.navigate(Routes.Search)
                  },
                  {
                    icon: 'notifications',
                    onPress: () => navigation.navigate(Routes.Notifications)
                  }
                ]}
              />
            );
          }
        })}
        name={Routes.RecommendedClubWorkouts}
        component={RecommendedClubWorkouts}
      />
      <Stack.Screen
        options={({ navigation }) => ({
          headerRight: () => {
            return (
              <HeaderContainer
                route={Routes.OnDemandClubWorkouts}
                items={[
                  {
                    icon: 'search',
                    onPress: () => navigation.navigate(Routes.Search)
                  },
                  {
                    icon: 'notifications',
                    onPress: () => navigation.navigate(Routes.Notifications)
                  }
                ]}
              />
            );
          }
        })}
        name={Routes.OnDemandClubWorkouts}
        component={OnDemandClubWorkouts}
      />
    </Stack.Navigator>
  );
}

export default function Router() {
  const {
    ui: { backgroundColor, barStyle },
    walkThrough: { hasBeenDone }
  } = useSelector((state) => state.appModel);
  const dispatch: Dispatch = useDispatch();

  useEffect(() => {
    if (hasBeenDone === false)
      dispatch.modalModel.setInfo({
        visible: true,
        title: formatMessage('Discover the basic-fit app'),
        subtitle: formatMessage(
          'Find out how you can start a workout, make a reservation, track your progress, and get inspired to Go For It!'
        ),
        acceptText: formatMessage('Show me'),
        onAccept: () => {
          dispatch.modalModel.setInfo({ visible: false });
          dispatch.appModel.setWalkThrough({ index: 0, visible: true });
        },
        closeText: formatMessage('Not now'),
        onClose: () => {
          dispatch.appModel.finishWalkThrough();
        }
      });
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [hasBeenDone]);

  useEffect(() => {
    const init = async () => {
      await dispatch.appModel.checkIfWalkThroughIsDone();
      await dispatch.authModel.refreshAuthedAxios();
      await dispatch.memberModel.getMemberInfo();
    };
    init();
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, []);

  return (
    <>
      <SafeAreaView style={[styles.safeAreaContainer, { backgroundColor: backgroundColor }]}>
        <StatusBar barStyle={barStyle} />
        <RouterComponent />
      </SafeAreaView>
    </>
  );
}
