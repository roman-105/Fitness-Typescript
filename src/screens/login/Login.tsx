import React, { useState } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { useNavigation } from '@react-navigation/native';
import { View, TouchableOpacity, ScrollView } from 'react-native';
import FastImage from 'react-native-fast-image';
import formatMessage from 'format-message';
import LoginHeroImg from '../../assets/images/login/login.jpg';
import { KeyboardAwareScrollView } from 'react-native-keyboard-aware-scroll-view';
import Typography from '../../components/Typography';
import BFButton from '../../components/Button/BFButton';
import BFTextInput, { BFTextInputPassword } from '../../components/TextInput/BFTextInput';
import styles from './Login-style';
import { Routes } from '../../router/routes';
import BFAnalytics from '../../utils/tools/analytics';
import { NO_PERSONALISATION_ERROR } from '../../store/models/personalisation/personalisationModelAdapter';

const Login = () => {
  const [user, setUser] = useState('gymtest2020+be-comfort@gmail.com');
  const [pass, setPass] = useState('2000And23');
  const [failed, setFailed] = useState(false);
  const dispatch: Dispatch = useDispatch();
  const navigation = useNavigation();

  const {
    loading: {
      effects: {
        authModel: { Login: isLoading }
      }
    }
  } = useSelector((state) => state);

  async function tryToLog() {
    BFAnalytics.logEvent('submitLogin');
    setFailed(false);
    try {
      await dispatch.authModel.Login({ user, pass });
    } catch (e) {
      BFAnalytics.logEvent('triggerError', {
        view: 'login',
        type: 'authenticationCredentialsError'
      });
      return setFailed(true);
    }

    try {
      await dispatch.personalisationModel.getPersonalisation();
    } catch (err) {
      if (err.message === NO_PERSONALISATION_ERROR) {
        return navigation.navigate(Routes.Consent);
      }
    }

    navigation.reset({ index: 0, routes: [{ name: Routes.Main }] });
  }

  function handleSignUp() {
    BFAnalytics.logEvent('tapSignupLink');
    navigation.navigate(Routes.SignUp);
  }

  const forgotUrl = 'https://my.basic-fit.com/new-password-request';

  function handleForgot() {
    BFAnalytics.logEvent('tapPasswordRecoveryLink');
    const setEmailJS = `(function() {
      setTimeout(() => {
        document.getElementsByTagName('input')[0].value = '${user}'
      }, 1000)
    })();`;
    navigation.navigate(Routes.Webview, { uri: forgotUrl, injectedJS: setEmailJS });
  }

  return (
    <>
      <ScrollView>
        <KeyboardAwareScrollView
          contentContainerStyle={styles.container}
          keyboardDismissMode="on-drag"
          testID="Login"
        >
          <View style={styles.imageContainer}>
            <FastImage source={LoginHeroImg} style={styles.image} />
          </View>
          <View style={[styles.padding, styles.spacebetween]}>
            <View>
              <Typography type="h1" fontFamily="trebleHeavy" style={styles.header}>
                {formatMessage('Login')}
              </Typography>
              <Typography type="regularbfa">
                {formatMessage('Use your Basic-Fit member details to login.')}
              </Typography>
              <Typography type="regularbfa">
                {formatMessage('And remember to go for it!')}
              </Typography>
              <View style={styles.inputs}>
                <BFTextInput value={user} onChangeText={setUser} placeholder="Username" />
                <BFTextInputPassword value={pass} onChangeText={setPass} placeholder="Password" />
              </View>
              {failed && (
                <Typography style={styles.errorText} type="regularbfa">
                  {formatMessage(
                    'Either the email address or password you provided is incorrect. Please try again.'
                  )}
                </Typography>
              )}
              <TouchableOpacity onPress={handleForgot}>
                <Typography style={styles.alignEnd} type="regularbfa">
                  {formatMessage('Forgot password')}
                </Typography>
              </TouchableOpacity>
            </View>
          </View>
        </KeyboardAwareScrollView>
      </ScrollView>
      <View style={styles.padding}>
        <BFButton
          disabled={pass.length < 3 || user.length < 1 || isLoading}
          onPress={tryToLog}
          title={formatMessage('LOGIN')}
        />
        <BFButton onPress={handleSignUp} white title={formatMessage('SIGN UP')} />
      </View>
    </>
  );
};

export default Login;
