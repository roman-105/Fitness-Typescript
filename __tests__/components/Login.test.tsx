import React from 'react';
import { fireEvent, render } from '@testing-library/react-native';
import Login from '../../src/screens/login/Login';
import * as reactRedux from 'react-redux';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import { useNavigation } from '@react-navigation/native';
import { Routes } from '../../src/router/routes';

let mockNavigate = jest.fn();
jest.mock('@react-navigation/native', () => {
  const actualNav = jest.requireActual('@react-navigation/native');
  return {
    ...actualNav,
    useNavigation: () => ({
      dangerouslyGetState: () => ({ routes: [] }),
      navigate: mockNavigate,
      setOptions: () => {},
      reset: () => {}
    })
  };
});

const mockStore = configureStore([]);
let store = mockStore({});

describe('Login tests', () => {
  beforeEach(() => {
    mockNavigate = jest.fn();
    const mockedLogin = jest.fn();
    const mockedIsLoading = false;
    jest.useFakeTimers();
    store = mockStore({});
    // @ts-ignore
    jest.spyOn(reactRedux, 'useDispatch').mockImplementation(() => ({
      authModel: {
        Login: mockedLogin
      }
    }));
    jest.spyOn(reactRedux, 'useSelector').mockImplementation(() => ({
      loading: {
        effects: {
          authModel: { Login: mockedIsLoading }
        }
      }
    }));
  });

  it('should render just fine', async () => {
    const { toJSON } = render(
      <Provider store={store}>
        <Login />
      </Provider>
    );
    expect(toJSON()).toMatchSnapshot();
  });

  it('should have login button disabled', async () => {
    const { getByText } = render(
      <Provider store={store}>
        <Login />
      </Provider>
    );
    const button = getByText('LOGIN');
    await fireEvent.press(button);
    const dispatch = reactRedux.useDispatch();
    expect(dispatch.authModel.Login).not.toBeCalled();
  });

  it('should have login button enabled', async () => {
    const { getByText, getByTestId } = render(
      <Provider store={store}>
        <Login />
      </Provider>
    );
    const user = getByTestId('Username');
    const password = getByTestId('Password');
    const button = getByText('LOGIN');
    await fireEvent.changeText(user, 'mockeUser123');
    await fireEvent.changeText(password, 'mockuser123');
    await fireEvent.press(button);
    const dispatch = reactRedux.useDispatch();
    expect(dispatch.authModel.Login).toBeCalledWith({
      pass: 'mockuser123',
      user: 'mockeUser123'
    });
  });

  it('should go to signin page on sigin tapped', async () => {
    const { getByText } = render(
      <Provider store={store}>
        <Login />
      </Provider>
    );
    const button = getByText('SIGN UP');
    await fireEvent.press(button);
    const navigation = useNavigation();
    expect(navigation.navigate).toBeCalledWith(Routes.SignUp);
  });

  it('should go to webview on forgot passw tapped', async () => {
    const { getByText } = render(
      <Provider store={store}>
        <Login />
      </Provider>
    );
    const button = getByText('Forgot password');
    await fireEvent.press(button);
    const navigation = useNavigation();
    expect(navigation.navigate).toBeCalledWith('WEBVIEW', {
      uri: 'https://my.basic-fit.com/new-password-request',
      injectedJS: expect.any(String)
    });
  });
});
