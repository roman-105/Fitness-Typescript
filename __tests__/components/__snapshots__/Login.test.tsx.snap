// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login tests should render just fine 1`] = `
Array [
  <RCTScrollView>
    <View>
      <RCTScrollView
        automaticallyAdjustContentInsets={false}
        contentContainerStyle={
          Object {
            "backgroundColor": "#fff",
            "flex": 1,
          }
        }
        contentInset={
          Object {
            "bottom": 0,
          }
        }
        enableAutomaticScroll={true}
        enableOnAndroid={false}
        enableResetScrollToCoords={true}
        extraHeight={75}
        extraScrollHeight={0}
        getScrollResponder={[Function]}
        handleOnScroll={[Function]}
        keyboardDismissMode="on-drag"
        keyboardOpeningTime={250}
        keyboardSpace={0}
        onScroll={[Function]}
        resetKeyboardSpace={[Function]}
        scrollEventThrottle={1}
        scrollForExtraHeightOnAndroid={[Function]}
        scrollIntoView={[Function]}
        scrollToEnd={[Function]}
        scrollToFocusedInput={[Function]}
        scrollToPosition={[Function]}
        showsVerticalScrollIndicator={true}
        testID="Login"
        update={[Function]}
        viewIsInsideTabBar={false}
      >
        <View>
          <View
            style={
              Object {
                "height": 185,
              }
            }
          >
            <View
              style={
                Array [
                  Object {
                    "overflow": "hidden",
                  },
                  Object {
                    "height": "100%",
                    "width": "100%",
                  },
                ]
              }
            >
              <FastImageView
                resizeMode="cover"
                source={null}
                style={
                  Object {
                    "bottom": 0,
                    "left": 0,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                  }
                }
              />
            </View>
          </View>
          <View
            style={
              Array [
                Object {
                  "paddingBottom": 8,
                  "paddingHorizontal": 20,
                },
                Object {
                  "flex": 1,
                  "justifyContent": "space-between",
                },
              ]
            }
          >
            <View>
              <Text
                style={
                  Array [
                    Object {
                      "fontFamily": "HeadingProTreble-Heavy",
                      "fontSize": 32,
                      "fontStyle": "normal",
                      "lineHeight": 38,
                      "textTransform": "uppercase",
                    },
                    Object {
                      "marginBottom": 8,
                      "marginTop": 16,
                    },
                    undefined,
                    undefined,
                    undefined,
                    undefined,
                    undefined,
                  ]
                }
              >
                Login
              </Text>
              <Text
                style={
                  Array [
                    Object {
                      "fontFamily": "HeadingProDouble-Regular",
                      "fontSize": 14,
                      "fontStyle": "normal",
                      "fontWeight": "500",
                      "letterSpacing": 0.1,
                      "lineHeight": 22,
                    },
                    undefined,
                    undefined,
                    undefined,
                    undefined,
                    undefined,
                    undefined,
                  ]
                }
              >
                Use your Basic-Fit member details to login.
              </Text>
              <Text
                style={
                  Array [
                    Object {
                      "fontFamily": "HeadingProDouble-Regular",
                      "fontSize": 14,
                      "fontStyle": "normal",
                      "fontWeight": "500",
                      "letterSpacing": 0.1,
                      "lineHeight": 22,
                    },
                    undefined,
                    undefined,
                    undefined,
                    undefined,
                    undefined,
                    undefined,
                  ]
                }
              >
                And remember to go for it!
              </Text>
              <View
                style={
                  Object {
                    "marginBottom": 10,
                    "marginTop": 24,
                  }
                }
              >
                <TextInput
                  allowFontScaling={true}
                  onChangeText={[Function]}
                  placeholder="Username"
                  placeholderTextColor="#31313688"
                  rejectResponderTermination={true}
                  style={
                    Array [
                      Object {
                        "alignItems": "center",
                        "backgroundColor": "#ECE9E8",
                        "flexDirection": "row",
                        "fontFamily": "HeadingProDouble-Regular",
                        "fontSize": 14,
                        "height": 52,
                        "margin": 1,
                      },
                      Object {
                        "paddingHorizontal": 16,
                      },
                      undefined,
                    ]
                  }
                  testID="Username"
                  underlineColorAndroid="transparent"
                  value=""
                />
                <View
                  style={
                    Object {
                      "alignItems": "center",
                      "backgroundColor": "#ECE9E8",
                      "flexDirection": "row",
                      "fontFamily": "HeadingProDouble-Regular",
                      "fontSize": 14,
                      "height": 52,
                      "margin": 1,
                    }
                  }
                >
                  <TextInput
                    allowFontScaling={true}
                    onChangeText={[Function]}
                    placeholder="Password"
                    placeholderTextColor="#31313688"
                    rejectResponderTermination={true}
                    secureTextEntry={true}
                    style={
                      Array [
                        Object {
                          "flex": 1,
                          "fontFamily": "HeadingProDouble-Regular",
                          "fontSize": 14,
                        },
                        Object {
                          "paddingHorizontal": 16,
                        },
                        undefined,
                      ]
                    }
                    testID="Password"
                    underlineColorAndroid="transparent"
                    value=""
                  />
                  <View
                    accessible={true}
                    focusable={true}
                    onClick={[Function]}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    style={
                      Object {
                        "opacity": 1,
                        "paddingHorizontal": 16,
                      }
                    }
                  >
                    <Text
                      style={
                        Array [
                          Object {
                            "fontFamily": "HeadingProDouble-Regular",
                            "fontSize": 14,
                            "fontStyle": "normal",
                            "fontWeight": "500",
                            "letterSpacing": 0.1,
                            "lineHeight": 22,
                            "textAlign": "right",
                          },
                          undefined,
                          undefined,
                          undefined,
                          undefined,
                          undefined,
                          undefined,
                        ]
                      }
                    >
                      Show
                    </Text>
                  </View>
                </View>
              </View>
              <View
                accessible={true}
                focusable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "opacity": 1,
                  }
                }
              >
                <Text
                  style={
                    Array [
                      Object {
                        "fontFamily": "HeadingProDouble-Regular",
                        "fontSize": 14,
                        "fontStyle": "normal",
                        "fontWeight": "500",
                        "letterSpacing": 0.1,
                        "lineHeight": 22,
                      },
                      Object {
                        "alignSelf": "flex-end",
                        "marginBottom": 16,
                      },
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                    ]
                  }
                >
                  Forgot password
                </Text>
              </View>
            </View>
          </View>
        </View>
      </RCTScrollView>
    </View>
  </RCTScrollView>,
  <View
    style={
      Object {
        "paddingBottom": 8,
        "paddingHorizontal": 20,
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#7322C9",
          "justifyContent": "center",
          "margin": 0,
          "opacity": 0.5,
          "padding": 8,
        }
      }
      testID="LOGIN"
    >
      <Text
        style={
          Array [
            Object {
              "fontFamily": "HeadingProTreble-Regular",
              "fontSize": 32,
              "fontStyle": "normal",
              "lineHeight": 38,
              "textTransform": "uppercase",
            },
            Array [
              Object {
                "color": "#FFFFFF",
                "fontSize": 16,
              },
              undefined,
            ],
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
          ]
        }
      >
        LOGIN
      </Text>
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#FFFFFF",
          "justifyContent": "center",
          "margin": 0,
          "opacity": 1,
          "padding": 8,
        }
      }
      testID="SIGN UP"
    >
      <Text
        style={
          Array [
            Object {
              "fontFamily": "HeadingProTreble-Regular",
              "fontSize": 32,
              "fontStyle": "normal",
              "lineHeight": 38,
              "textTransform": "uppercase",
            },
            Array [
              Object {
                "color": "#FFFFFF",
                "fontSize": 16,
              },
              Object {
                "color": "#313136",
              },
            ],
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
          ]
        }
      >
        SIGN UP
      </Text>
    </View>
  </View>,
]
`;
